File: backend/app/schemas/follow.py
"""
关注相关的数据模型
"""
from pydantic import BaseModel
from typing import Optional, List


class FollowUserRequest(BaseModel):
    """关注用户请求"""
    followerId: int
    targetUserId: int


class FollowBrandRequest(BaseModel):
    """关注品牌请求"""
    userId: int
    brandId: int


class BatchFollowBrandsRequest(BaseModel):
    """批量关注品牌请求"""
    userId: int
    brandIds: List[int]


class FollowingUser(BaseModel):
    """关注的用户信息"""
    userId: int
    username: str
    avatar: str = ""
    bio: str = ""
    location: str = ""


class FollowingBrand(BaseModel):
    """关注的品牌信息"""
    brandId: int
    name: str
    category: str = ""
    coverImage: str = ""
    country: str = ""
    followersCount: int = 0

File: web/src/lib/services/follow.ts
/**
 * Follow service (web).
 *
 * Mirror of [frontend/src/services/followService.ts](../../../frontend/src/services/followService.ts)
 * for the endpoints the web currently needs. Brand + mutual follow endpoints
 * are included up-front so /archive/brands and /me/follows can use them later
 * without another service file.
 */

import { apiClient } from "../api-client";

export interface FollowingUser {
  userId: number;
  username: string;
  avatar: string;
  bio: string;
  location: string;
}

export interface FollowingBrand {
  brandId: number;
  name: string;
  category: string;
  coverImage: string;
  country: string;
  followersCount: number;
}

export const followService = {
  // --- user follow ---
  followUser: (followerId: number, targetUserId: number) =>
    apiClient.post<void>("/api/follow/user", { followerId, targetUserId }),
  unfollowUser: (followerId: number, targetUserId: number) =>
    apiClient.delete<void>("/api/follow/user", {
      body: { followerId, targetUserId },
    }),
  isFollowingUser: (followerId: number, targetUserId: number) =>
    apiClient.get<boolean>(
      `/api/follow/user/${followerId}/is-following/${targetUserId}`,
    ),
  getFollowersCount: (userId: number | string) =>
    apiClient.get<number>(`/api/follow/user/${userId}/followers/count`),
  getFollowingCount: (userId: number | string) =>
    apiClient.get<number>(`/api/follow/user/${userId}/following/count`),
  getFollowingUsers: (userId: number | string) =>
    apiClient.get<FollowingUser[]>(
      `/api/follow/users/${userId}/following-users`,
    ),
  getFollowers: (userId: number | string) =>
    apiClient.get<FollowingUser[]>(`/api/follow/users/${userId}/followers`),

  // --- brand follow ---
  followBrand: (userId: number, brandId: number) =>
    apiClient.post<void>("/api/follow/brand", { userId, brandId }),
  unfollowBrand: (userId: number, brandId: number) =>
    apiClient.delete<void>("/api/follow/brand", {
      body: { userId, brandId },
    }),
  isFollowingBrand: (userId: number, brandId: number) =>
    apiClient.get<boolean>(
      `/api/follow/user/${userId}/is-following-brand/${brandId}`,
    ),
  getFollowingBrands: (userId: number | string) =>
    apiClient.get<FollowingBrand[]>(
      `/api/follow/users/${userId}/following-brands`,
    ),
  getBrandFollowersCount: (brandId: number | string) =>
    apiClient.get<number>(`/api/follow/brand/${brandId}/followers/count`),
};
